### ISSUE TITLE
Render list items lazily

### ISSUE THREAD

**Issue by pigulla** [jquense/react-widgets#97]
I'm making extensive use of react-widgets in my project, and they work wonderfully. However, on one page I need to render a few dozen dropdowns, each of which has several dozen items using custom `itemComponent`s and `groupComponent`s.

Unfortunately, the initial rendering has become a problem. Once everything is done it's blazingly fast but currently it takes almost a second for that page to show up.

Is there a way to make the `DropdownList` and `SelectList` render its items on demand rather than eagerly?

**Comment by jquense** [comment]
hmm. that's a good question. I would imagine the issue is probably due to how all popup are initially rendered opened open (then closed) to handle needed popup animation calculations. so the browser has to render it all on screen at once. 

I might be able to optimize that internally (though your use case is a bit out of the ordinary :P) but in the meantime I would try wrapping the DropdownList and passing an empty array for `data` and then switching to the real `data' at the first onFocus event. That way the lists will all start with nothing. SelectLists are harder because the items are all visible already so I'm not sure how to handle that generally but you could use the same trick for when the list scrolls into view perhaps...

**Comment by pigulla** [comment]
Thanks, that wrapper-idea solved the problem for now. It's still be nice to have this supported natively :-)

**Comment by jquense** [comment]
glad that helped! I'll take a look into adding it, I need to do a bit a of poking first to see if the lag is mostly paint times, or whether React is choking on creating all the `ReactElements` (or both)

**Comment by jquense** [comment]
@pigulla so good news it looks like this is the sort of thing that I can pretty easily incorporate. I'm working on a few obvious improvements to the `Popup` component to make sure it doesn't do unnecessary painting/DOM calculation. I also think that I can lazily render the entire contents of them (without breaking functionality) on focus which seems to help quite a bit, as well. Look for it in an upcoming release

**Comment by pigulla** [comment]
Awesome. Let me see what feature request I can come up with next ;)

**Comment by reod** [comment]
I had similar problem as @pigulla, but with `DateTimePicker`. It would be awesome to have `Popup` lazily rendered. I'm looking forward for update ;)

**Comment by jquense** [comment]
the newest release, `2.6.0`, contains this optimization.

**Comment by reod** [comment]
It work like a charm, performance is significant better when rendering 20+ `DateTimePicker`a. Thx.
